crates/nargo_cli/tests/test_data/strings/src/main.nr

```diff
@@ -39,16 +39,26 @@ fn test_prints_strings() {
 fn test_prints_array() {
     let array = [1, 2, 3, 5, 8];
 
-    // TODO: Printing structs currently not supported
-    // let s = Test { a: 1, b: 2, c: [3, 4] };
-    // std::println(s);
+    let s = Test { a: 1, b: 2, c: [3, 4] };
+    std::println(s);
 
     std::println(array);
 
     let hash = std::hash::pedersen(array);
     std::println(hash);
 }
 
+fn inner(hex_as_field: Field) {
+    assert(hex_as_field != 0x41);
+    let hash = std::hash::pedersen([hex_as_field]);
```

Comment:
Is this use of pedersen in order to create a non-comptime value to be printed?

Once pedersen is rust-solvable then with https://github.com/noir-lang/noir/pull/1981 this won't be true anymore.

Reply:
It is comptime as it is only used in a test. I was only using this for my own local testing I will update this method before converting the PR back to ready to review

Reply:
Ah, I think comptime was slightly wrong terminology. While we know that the pedersen output is fixed at comptime, we don't know a constant pair of fields which will be equal to the pedersen hash output as the compiler has no way to execute the black box function even with all the inputs.

Reply:
Ah I see. Looking back at the original test I believe printing the result of pedersen was included for this reason, however, `println` and the black box func implementations were both quite different when the original println is added. Printing of witness values and black box func output are both supported in a much simpler manner by moving printing to execution.